File: Leet_Weekly_Contest_279/Hoovely/Leet_2164.py
# leetcode 2164
# 홀수 인덱스 => 내림차순 정렬
# 짝수 인덱스 => 오름차순 정렬

class Solution:
    def sortEvenOdd(self, nums: List[int]) -> List[int]:
        if len(nums) <= 2:
            return nums

        odd_nums = []  # 홀수
        even_nums = []  # 짝수
        for i in range(len(nums)):
            if i % 2 == 0:
                even_nums.append(nums[i])
            else:
                odd_nums.append(nums[i])
        odd_nums.sort(reverse=True)
        even_nums.sort()

        for i in range(len(nums)):
            if i % 2 == 0:
                nums[i] = even_nums[i//2]
            else:
                nums[i] = odd_nums[i//2]

        return nums

File: Leet_Weekly_Contest_279/red0825/LEET_2164.cpp
class Solution {
public:
    vector<int> sortEvenOdd(vector<int>& nums) {
        vector<int> Even;
        vector<int> Even_index;
        vector<int> Odd;
        vector<int> Odd_index;
        int i = 0;
        for(auto x : nums){
            if(i % 2 == 0){
                Even.push_back(x);
                Even_index.push_back(i);
            }
            else{
                Odd.push_back(x);
                Odd_index.push_back(i);
            }
            i++;
        }
        sort(Even.begin(), Even.end());
        sort(Odd.begin(), Odd.end(), greater<int>());
        i = 0;
        for(auto x : Even_index){
            nums[x] = Even[i];
            i++;
        }
        i = 0;
        for(auto x : Odd_index){
            nums[x] = Odd[i];
            i++;
        }
        return nums;
    }
};
